src/screen.rs

```diff
@@ -1057,19 +1089,21 @@
         let term = term.as_ref();
 
         // Output the left prompt if it has changed.
+        let osc_133_prompt_start =
+            |zelf: &mut Screen| zelf.write_bytes(b"\x1b]133;A;click_events=1\x07");
         if zelf.scrolled() && !is_final_rendering {
             zelf.r#move(0, 0);
             zelf.outp
                 .borrow_mut()
                 .tputs_if_some(&term.and_then(|term| term.clr_eol.as_ref()));
             zelf.actual_left_prompt.clear();
             zelf.actual.cursor.x = 0;
-        } else if left_prompt != zelf.actual_left_prompt || (zelf.scrolled() && is_final_rendering)
+        } else if left_prompt != zelf.actual_left_prompt
+            || left_prompt.is_empty()
```

Comment:
is this because `actual_left_prompt` is initialized as empty (and probably reset after every command)?
If true, wouldn't it be more obvious to make it an `Option<WString>`? 
Since `function fish_prompt; end` is valid (which seems fine), so we want to properly distinguish between "empty" and "uninitialized".